client/settings/payment-settings/account-keys-modal.js

```diff
@@ -189,6 +189,25 @@ export const AccountKeysModal = ( {
 			const { name, value } = curr;
 			return { ...acc, [ name ]: value };
 		}, {} );
+
+		// If we are deleting keys for this mode and there are no other keys set, we need to reload to render the connect page.
+		const savingEmptyKeys =
+			! keysToSave.publishable_key &&
+			! keysToSave.secret_key &&
+			! keysToSave.test_publishable_key &&
+			! keysToSave.test_secret_key;
+		const noLiveKeysSaved =
+			! accountKeys.publishable_key && ! accountKeys.secret_key;
+		const noTestKeysSaved =
+			! accountKeys.test_publishable_key && ! accountKeys.test_secret_key;
+		if (
+			savingEmptyKeys &&
+			( ( testMode && noLiveKeysSaved ) ||
+				( ! testMode && noTestKeysSaved ) )
+		) {
+			forcePageReloadOnSave = true;
```

Comment:
In the Settings page is still closing the Modal before redirecting, it works fine in the Connect page and when opening the modal from the dropdown menu in the Account Details section.

I'll look at it again tomorrow 😞